existential-vs-universal-quantification/article.md

```diff
@@ -0,0 +1,194 @@
+# Existential vs universal quantification
+
+## Introduction
+This article gives brief information about one of the most common extension in Haskell projects - `ExplicitForAll`
+and syntax constructions that this extension brings to language.
+Stay tuned if you want to know how to create lists of differently typed variables, how to add complex constraints and more!
+
+From type theory we know about the existence of two quantifiers - `$\forall$` and `$\exists$`.
+The former means: `"For every term x the statement is true"`, the later: `"There is at least one term x for which this statement is true"`.
+
+Since Haskell is based on type theory there is a place for these quantifiers in language syntax, however, in Haskell we are talking not about some terms and statements, but about types.
+By default, all type variables are bound by a `forall` quantifier, which means that every type, that passes constraints, can be used inside this function/type declaration.
+
+```haskell
+id :: forall a. a -> a
+id x = x
+```
+
+You can't see this quantification without using a specific extension - "Explicit Forall".
+
+## Universal quantification and ExplicitForall
```

Comment:
The `Universal quantification and ExplicitForAll` section seems a bit unstructured and doesn't seem to have a central narrative.
Let's write down a quick summary of what we have:

* Introduction: a single sentence explaining what universal quantification is + an example (`id`)
  * Universal quantification and ExplicitForAll: an example showing a possible use case for `ExplicitForAll` (`prepend`)
    * Usage: an example explaining the syntax (`func`)
    * Examples: 2 more examples showing more use cases for `ExplicitForAll` again

Here are some suggestions:

1. As stated in [my other comment](https://github.com/serokell/blog-posts/pull/4/files#r782087623), the sentence explaining what universal quantification is should be in the the `Universal quantification and ExplicitForAll` section, not in the introduction.
1. The `Universal quantification and ExplicitForAll` section should focus more on _what_ universal quantification is, and when and why it's useful, and also mention _who_'s responsible for instantiating universally quantified type variables (see [my comment here](https://github.com/serokell/blog-posts/pull/4/files#r782085135)). I think it's important to establish that _before_ mentioning that (1) there's an implicit `forall` and (2) it can be made explicit. The subsections can then go into detail about _why_ and _when_ a user would want to make it explicit.
1. We're explaining the syntax near the end of the section, on our 3rd example, `func`. By this point, the reader is already familiar with the syntax.
   So maybe we should explain the syntax before diving into complex examples. 
1. Looks like the goal of the `prepend` example is to show _when_ making the quantifier explicit is needed. 
   This is the same goal as `veryLongFunction` and `monadicFunction`, so looks like `prepend` belongs in the `Examples` section.

Reply:
Seems like the rewrite addresses most of this comment so I'll resolve it